core/src/main/java/hudson/security/AuthenticationProcessingFilter2.java

```diff
@@ -100,26 +116,25 @@ protected void onSuccessfulAuthentication(HttpServletRequest request, HttpServle
             newSession.setAttribute(UserSeedProperty.USER_SESSION_SEED, sessionSeed);
         }
 
-        // as the request comes from Acegi redirect, that's not a Stapler one
+        // as the request comes from Spring Security redirect, that's not a Stapler one
         // thus it's not possible to retrieve it in the SecurityListener in that case
         // for that reason we need to keep the above code that apply quite the same logic as UserSeedSecurityListener
         SecurityListener.fireLoggedIn(authResult.getName());
     }
 
     /**
      * Leave the information about login failure.
-     *
-     * <p>
-     * Otherwise it seems like Acegi doesn't really leave the detail of the failure anywhere.
      */
     @Override
-    protected void onUnsuccessfulAuthentication(HttpServletRequest request, HttpServletResponse response, AuthenticationException failed) throws IOException {
-        super.onUnsuccessfulAuthentication(request, response, failed);
+    protected void unsuccessfulAuthentication(HttpServletRequest request, HttpServletResponse response, AuthenticationException failed) throws IOException, ServletException {
+        super.unsuccessfulAuthentication(request, response, failed);
         LOGGER.log(Level.FINE, "Login attempt failed", failed);
+        /* TODO this information appears to have been deliberately removed from Spring Security:
         Authentication auth = failed.getAuthentication();
         if (auth != null) {
             SecurityListener.fireFailedToLogIn(auth.getName());
```

Comment:
Right. I spent a bit of time trying to figure out how to restore this, without success. If you find something, I guess it can be added later.

In terms of impact, I can find only https://github.com/jenkinsci/datadog-plugin/blob/6d327f9e6719ee01689e10410b627b39958c7fad/src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogSecurityListener.java#L131-L134 and https://github.com/jenkinsci/jenkins/blob/f04114c0f813501da9a072b984a6cf2f2f0a2180/core/src/main/java/hudson/security/AbstractPasswordBasedSecurityRealm.java#L101 still fires that event.